sdk/keyvault/Azure.Security.KeyVault.Keys/src/DeletedKey.cs

```diff
@@ -3,27 +3,66 @@
 // license information.
 
 using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
+using System.Text.Json;
 
 namespace Azure.Security.KeyVault.Keys
 {
     public class DeletedKey : Key
     {
         public string RecoveryId { get; private set; }
 
-        public DateTime? DeletedDate { get; private set; }
+        public DateTimeOffset? DeletedDate { get; private set; }
 
-        public DateTime? ScheduledPurgeDate { get; private set; }
+        public DateTimeOffset? ScheduledPurgeDate { get; private set; }
 
-        public DeletedKey(string name, string recoveryId, DateTime? deletedDate, DateTime? scheduledPurge)
+        public DeletedKey(string name) : base(name) { }
+
+        public DeletedKey (string name, string recoveryId, DateTimeOffset? deletedDate, DateTimeOffset? scheduledPurge)
             : base(name)
         {
             RecoveryId = recoveryId;
             DeletedDate = deletedDate;
             ScheduledPurgeDate = scheduledPurge;
         }
+
+        internal override void WriteProperties(ref Utf8JsonWriter json)
+        {
+            base.WriteProperties(ref json);
+
+            if (RecoveryId != null)
+            {
+                json.WriteString("recoveryId", RecoveryId);
+            }
+
+            if (DeletedDate.HasValue)
+            {
+                json.WriteNumber("deletedDate", DeletedDate.Value.ToUnixTimeMilliseconds());
+            }
+
+            if (ScheduledPurgeDate.HasValue)
+            {
+                json.WriteNumber("scheduledPurgeDate", ScheduledPurgeDate.Value.ToUnixTimeMilliseconds());
+            }
+        }
+
+        internal override void ReadProperties(JsonElement json)
+        {
+            base.ReadProperties(json);
+
+            if (json.TryGetProperty("recoveryId", out JsonElement recoveryId))
+            {
+                RecoveryId = recoveryId.GetString();
+            }
+
+            if (json.TryGetProperty("deletedDate", out JsonElement deletedDate))
+            {
+                DeletedDate = DateTimeOffset.FromUnixTimeMilliseconds(deletedDate.GetInt64());
+            }
+
+            if (json.TryGetProperty("scheduledPurgeDate", out JsonElement scheduledPurgeDate))
```

Comment:
TryGetProperty is an O(N) lookup.  Consider using EnumerateObject and matching inside a loop.  (With Preview7 you could then also use the more efficient NameEquals method (more efficient than reading the Name property).

```C#
foreach (JsonProperty prop in json.EnumerateObject())
{
    switch (prop.Name)
    {
        case "recoveryId":
            RecoveryId = prop.Value.GetString();
            break;
        case "deletedDate":
            DeletedDate = DateTimeOffset.FromUnixTimeMilliseconds(prop.Value.GetInt64());
            break;
        case "scheduledPurgeDate":
            ScheduledPurgeDate = DateTimeOffset.FromUnixTimeMilliseconds(prop.Value.GetInt64());
            break;
}
```

Reply:
Nice! Thank your Jeremy, I will update it!

Reply:
Issue https://github.com/Azure/azure-sdk-for-net/issues/6475 to use `NameEquals` once Preview 7 is available